### ISSUE TITLE
[Bug] CenterHead Regression Loss

### ISSUE THREAD

**Issue by Code-Gratefully** [open-mmlab/mmdetection3d#2912]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection3d/issues) and [Discussions](https://github.com/open-mmlab/mmdetection3d/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection3d.readthedocs.io/en/latest/notes/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (dev-1.x)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.x) or [latest version (dev-1.0)](https://github.com/open-mmlab/mmdetection3d/tree/dev-1.0).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmdetection3d

### Environment

N/A - See additional information below

### Reproduces the problem - code sample

N/A - See additional information below

### Reproduces the problem - command or script

N/A - See additional information below

### Reproduces the problem - error message

See additional information below

### Additional information

In computing the bounding box regression loss of CenterHead (https://github.com/open-mmlab/mmdetection3d/blob/main/mmdet3d/models/dense_heads/centerpoint_head.py#L609) , there is no association between preds_dict and batch_gt_instances_3d.  In other words, the prediction and ground truth target seemed to be matched according to arbitrary order. The loss function provided in the default pytorch L1Loss or mmdet.L1loss does not seemed to have any association logic neither, so is this a bug or the association logic is somehow abstracted within other components of the framework?
